### PR TITLE
Exposing parameters for controlling the ekf fusion logic when the field is close to vertical

### PR DESCRIPTION
**Pull request by CarlOlsson**
@priseborough 
Exposes the functionality implemented [here](https://github.com/PX4/ecl/pull/281)

### CODE REVIEW

File: src/modules/ekf2/ekf2_main.cpp [hunk 1]
```diff
@@ -242,6 +242,8 @@ class Ekf2 : public control::SuperBlock
 	control::BlockParamExtInt _mag_fuse_type;         // integer ued to control the type of magnetometer fusion used
 	control::BlockParamExtFloat _mag_acc_gate;	// manoeuvre threshold for use of 3-axis fusion (m/s**2)
 	control::BlockParamExtFloat _mag_yaw_rate_gate;	// yaw rate threshold for use of 3-axis fusion (rad/s)
+	control::BlockParamExtFloat _mag_field_vertical;	// selectes how to fuse mag when the field is close to vertical
```
Comment (yyassa) [inline src/modules/ekf2/ekf2_main.cpp]: This guy should be an integer

**Comment by priseborough** [comment]
@CarlOlsson, the checks are failing.

**Comment by CarlOlsson** [comment]
@priseborough I think the checks are failing because this pr does not update the ecl reference, how do you propose to handle this?

**Comment by priseborough** [comment]
I rebased and pushed the new patches to here:

https://github.com/priseborough/Firmware/tree/ekfHighLatMag-wip.  However during SITL testing I was unable to view or modify the parameters. 

pxh> ERROR [lib__controllib] exceeded max params for block: EKF

Unfortunately it looks like we have now exceeded the parameter size limit for the EKF block. I will review the block and see what can be done to remove or combine parameters.

**Comment by dagar** [comment]
Can this be closed?

**Comment by priseborough** [comment]
Yes, the ecl PR this is required  has been replaced by https://github.com/PX4/ecl/pull/335 so the changes are no longer valid.

**Pull request closed without merging**